docs/articles/announcements/Towards-NUnit4.md

```diff
@@ -0,0 +1,201 @@
+
+## Proper async/await, you can now await the Asserts
+
+NUnit 3 has implemented async functionality using a pattern called 'sync-over-async.' This approach allowed for its implementation without requiring significant underlying changes.
+
+In version 4 proper async/await has been implemented with a series of new assert methods, `Assert.ThatAsync` which can be `await`'ed.
+
+```cs
+    [Test]
+    public async Task AssertionPasses_CompletedTaskWithResult_EqualsResult()
+    {
+        await Assert.ThatAsync(() => Task.FromResult(42), Is.EqualTo(42));
+    }
+```
+
+Version 4 also introduces a new feature called Assert.MultipleAsync, which allows you to mix async and sync asserts within the same block. This enables you to perform multiple assertions, both asynchronous and synchronous, in a more concise and streamlined manner. Moreover, it's important to note that Assert.MultipleAsync is awaitable, providing flexibility in handling asynchronous operations and assertions.
+
+```cs
+    [Test]
+    public async Task AssertMultipleAsyncSucceeds()
+    {
+        await Assert.MultipleAsync(async () =>
+        {
+            await Assert.ThatAsync(() => Task.FromResult(42), Is.EqualTo(42));
+            Assert.That("hello", Is.EqualTo("hello"));
+            await Assert.ThatAsync(() => Task.FromException(new  ArgumentNullException)), Throws.ArgumentNullException);
+        });
+    }
+```
+
+## Nullability
+
+Version 4 has implemented stricter nullability throughout the codebase. While there are still some areas that require fixing, overall, it should now conform to the nullability requirements.  See [3376](https://github.com/nunit/nunit/issues/3376) for details.
+
+These changes, along with the improved assert messages, have resulted in null values no longer being allowed for messages. As a consequence, code like the example below will not compile and will result in [CS0121](https://learn.microsoft.com/en-us/dotnet/csharp/misc/cs0121?f1url=%3FappId%3Droslyn%26k%3Dk(CS0121))):
+
+![AssertNull](~/images/AssertNull.png)
+It's important to update such code to use valid non-null message strings
+
+## Platform support
+
+The lowest framework platforms support in Version 4 are **.net framework 4.6.2** and **.net 6.0**.
+
+## Classic/Legacy asserts
+
+The classic/legacy asserts, like `Assert.AreEqual`, have now been moved into its own project, and will be released as a separate package.  They are now in the namespace `NUnit.Framework.Classic`.
+In the early V4 alpha version they will be delivered in the standard NUnit package.
+
+This means that the default assert library going forward will be the `Constraint based` asserts.
+
+Exactly how these methods will be used is not quite decided yet.  We will probably need to do some namechanges to avoid clashes.
+
+## Testing it out
+
+If you add our Myget source to your app's nuget feeds, you can try out the new features yourself.
```

Comment:
That might be defunct soon.
For work we use github packages, I presume it works with a public repository:
```
<add key="NuGetGitHub" value="https://nuget.pkg.github.com/nunit/index.json" />
```